sui/deploy-contract.js

```diff
@@ -0,0 +1,76 @@
+const { saveConfig, printInfo } = require('../evm/utils');
+const { Command, Argument } = require('commander');
+const { publishPackage, updateMoveToml } = require('@axelar-network/axelar-cgp-sui/scripts/publish-package');
+
+const { addBaseOptions } = require('./cli-utils');
+const { getWallet, printWalletInfo } = require('./sign-utils');
+const { loadSuiConfig, findPublishedObject } = require('./utils');
+
+// Add more contracts here to support more modules deployment
+const contractMap = {
+    gasService: {
+        packageName: 'gas_service',
+        contractName: 'GasService',
+        displayName: 'Gas Service',
+        chainConfigKey: 'axelar_gas_service',
+    },
+};
+
+async function processCommand(contractName, config, chain, options) {
+    const contract = contractMap[contractName];
+
+    const [keypair, client] = getWallet(chain, options);
+
+    await printWalletInfo(keypair, client, chain, options);
+
+    if (!chain.contracts[contract.chainConfigKey]) {
+        chain.contracts[contract.chainConfigKey] = {};
+    }
+
+    const published = await publishPackage(contract.packageName, client, keypair);
+    const packageId = published.packageId;
+
+    updateMoveToml(contract.packageName, packageId);
+
+    const contractObject = findPublishedObject(published, contract.packageName, contract.contractName);
+    const gasCollectorCapObject = findPublishedObject(published, contract.packageName, 'GasCollectorCap');
+
+    const contractConfig = chain.contracts[contract.chainConfigKey];
+    contractConfig.address = packageId;
+    contractConfig.objects = {
+        [contractName]: contractObject.objectId,
+    };
+
+    switch (contractName) {
+        case 'gasService':
+            contractConfig.objects.gasCollectorCap = gasCollectorCapObject.objectId;
+            break;
+        default:
+            break;
```

Comment:
error should be thrown here

Reply:
this line is impossible to reach since it'll throw earlier by the commander lib as available contract names are defined as choices.

Reply:
i understand but we shouldn't rely on that. For e.g. what if the choices are extended or renamed but this switch case isn't updated. It would be a less readable error